oap-server/ai-pipeline/src/main/proto/ai_http_uri_recognition.proto

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "org.apache.skywalking.oap.server.ai.pipeline.grpc";
+
+service HttpUriRecognitionService {
+    rpc fetchAllPatterns(HttpUriRecognitionSyncRequest) returns (HttpUriRecognitionResponse) {}
+    rpc feedRawData(HttpUriRecognitionRequest) returns (NoResponse) {}
+}
+
+message HttpUriRecognitionSyncRequest {
+    string service = 1;
+}
```

Comment:
If https://github.com/apache/skywalking/pull/10884#discussion_r1221205604 is able to support, then we could add version ID here and in `HttpUriRecognitionResponse`.
Then AI server doesn't need to respond all patterns if the version is the latest.